bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexApiConnector.java

```diff
@@ -198,28 +204,58 @@ public boolean getApi() {
                 }
             }
             return false;
-        } catch (IOException e) {
+        } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
             logger.debug("An exception occurred while fetching API :'{}'", e.getMessage(), e);
         }
         return false;
     }
 
     /**
-     * Make an HTTP request and return the class object that was used when calling.
+     * Make an HTTP request and return the response as a string.
      *
-     * @param <T> Class being used(dto)
      * @param method GET/POST
      * @param url What URL to call
      * @param headers Additional headers that will be used
-     * @param type class type for the XML parsing
-     * @return Returns a class object from the data returned by the call
+     * @param verify Flag to indicate if ssl certificate checking should be done
+     * @return Returns a string of the http response
      * @throws IOException
+     * @throws ExecutionException
+     * @throws TimeoutException
+     * @throws InterruptedException
      */
-    private <T> T doHttpRequest(String method, String url, Properties headers, Class<T> type) throws IOException {
-        String response = HttpUtil.executeUrl(method, url, headers, null, null, REQUEST_TIMEOUT_MS);
+    private String doHttpRequest(String method, String url, Properties headers, boolean verify)
+            throws IOException, InterruptedException, TimeoutException, ExecutionException {
+        final String response;
+        if (verify) {
+            // Requests sent to the PLEX.tv servers should use certificate checking
+            response = HttpUtil.executeUrl(method, url, headers, null, null, REQUEST_TIMEOUT_MS);
+        } else {
+            // Requests sent to the local server need to bypass certificate checking via the custom httpClient
+            final Request request = httpClient.newRequest(url).method(HttpUtil.createHttpMethod(method));
+            for (String httpHeaderKey : headers.stringPropertyNames()) {
+                if (httpHeaderKey.equalsIgnoreCase(HttpHeader.USER_AGENT.toString())) {
+                    request.agent(headers.getProperty(httpHeaderKey));
+                } else {
+                    request.header(httpHeaderKey, headers.getProperty(httpHeaderKey));
+                }
+            }
+            final ContentResponse res = request.send();
+            response = res.getContentAsString();
```

Comment:
@morph166955 I was working on this change to catch the failure and re-try with http
```java
        try {
                // Requests sent to the local server need to bypass certificate checking via the custom httpClient
                final Request request = httpClient.newRequest(url).method(HttpUtil.createHttpMethod(method));
                for (String httpHeaderKey : headers.stringPropertyNames()) {
                    if (httpHeaderKey.equalsIgnoreCase(HttpHeader.USER_AGENT.toString())) {
                        request.agent(headers.getProperty(httpHeaderKey));
                    } else {
                        request.header(httpHeaderKey, headers.getProperty(httpHeaderKey));
                    }
                }
                final ContentResponse res = request.send();
                response = res.getContentAsString();
            } catch (Exception e) {
                // if https conection to the local server fails, re-try with http
                response = HttpUtil.executeUrl(method, url.replace("https:", "http:"), headers, null, null, REQUEST_TIMEOUT_MS);
            }
```

Reply:
I really think this is a bad idea. We shouldn't fall back to insecure urls if connection fails. Instead we have to find out why it fails and fix the https connection

Reply:
Ok, I'll let you sort it out. It was just rather alarming that a problem report came in right after the first nightly build of this PR.

Reply:
Perhaps an advanced configuration option could allow users to explicitly use http rather than https? I agree it shouldn't silently fallback to http for security reasons. See also **useTLS** here: https://www.openhab.org/addons/bindings/lgwebos/#thing-configuration